src/environment/stella_environment.cpp

```diff
@@ -47,6 +47,17 @@ StellaEnvironment::StellaEnvironment(OSystem* osystem, RomSettings* settings)
   m_num_reset_steps = 4;
   m_cartridge_md5 = m_osystem->console().properties().get(Cartridge_MD5);
 
+  Logger::Info << "Random seed is "
+               << m_osystem->settings().getInt("random_seed") << std::endl;
+  // Initialize RNG
+  if (m_osystem->settings().getInt("random_seed") == -1) {
```

Comment:
We shouldn't set a random seed if the user-specified random seed is `0`. This is a common seed value that shouldn't be taken as a "random seed". Using -1 should help alleviate this.